src/shared/markdown.ts

```diff
@@ -74,73 +73,66 @@ const html5EmbedConfig = {
 };
 
 function localCommunityLinkParser(md) {
-  const pattern =
-    /(!\b[^@\s]+@[^@\s]+\.[^.\s]+\b)|\/c\/([^@\s]+)(@[^@\s]+\.[^.\s]+\b)?/g;
-
   md.core.ruler.push("replace-text", state => {
-    const tokens = state.tokens;
-
-    for (let i = 0; i < tokens.length; i++) {
-      if (tokens[i].type === "inline") {
-        const token = tokens[i];
-
-        const originalContent = token.content;
-
-        let lastIndex = 0;
-        originalContent.replace(
-          pattern,
-          (match, fullDomainMatch, name, domainTld, index) => {
-            let url;
-            // ex: !Testing@example.com
-            if (fullDomainMatch) {
-              const [name, domain, tld] = fullDomainMatch
-                .slice(1)
-                .split("@")
-                .join(".")
-                .split(".");
-              url = `${getHttpBase()}/c/${name}@${domain}.${tld}`;
+    /**
+     * Accepted formats:
+     * !community@server.com
+     * /c/community@server.com
+     * /m/community@server.com
+     * /u/username@server.com
+     */
+    const pattern =
+      /(\/[c|m|u]\/[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}|![a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})/g;
```

Comment:
Instead of having the whole regex pattern written twice, one for /_x_/ and one for !, could this be simplified to 
`/(\/[c|m|u]\/|!)[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}/g`

Reply:
This is great, thank you!